Terraform remote S3 state

jonmjoyce opened the pull request:
- Added backend configuration to point to S3 bucket
- If it doesn't exist, first create the S3 bucket under terraform/remote-state/ 
   - `terraform init`
   - `terraform apply`
- Then in /terraform, run `terraform init` again to initialize the new backend

patrick-tripp reviewed (changes requested):
Needs to generate a unique s3 key for each deployment.

jonmjoyce commented:
> Needs to generate a unique s3 key for each deployment.

I added instructions in the README to create a unique workspace before deploying resources. Terraform will automatically track the workspaces in the remote state file.

I also added instructions for initializing the remote state before running terraform init on the project.
